core/src/main/java/org/apache/carbondata/scan/scanner/impl/FilterScanner.java

```diff
@@ -151,7 +160,13 @@ private void fillScannedResult(BlocksChunkHolder blocksChunkHolder)
     for (int i = bitSet.nextSetBit(0); i >= 0; i = bitSet.nextSetBit(i + 1)) {
       indexes[index++] = i;
     }
-
+    // loading delete data cache in blockexecutioninfo instance
+    DeleteDeltaCacheLoaderIntf deleteCacheLoader =
```

Comment:
Scanner responsibility is to load to scan the data why it is loading delete cache ? it should be loaded before scanning like block loading if any delete delta cache exists